### ISSUE TITLE
`clippy::ptr_arg` wrongly warns against `&mut Vec<_>` argument in trait method

### ISSUE THREAD

**Issue by little-dude** [rust-lang/rust-clippy#8410]
### Summary

In nightly, the `ptr_arg` lint seems a bit aggressive and warns against the `&mut Vec<u8>` argument the following method:

```rust
trait Foo {
     fn foo(v: &mut Vec<u8>);
}
```

Note that it accepts:

```rust
fn foo(v: &mut Vec<u8>);
```

### Lint Name

ptr_arg

### Reproducer

I tried this code:

```rust
pub trait Foo {
    // Clippy suggests turning this into a &mut [u8]
    fn foo(&self, v: &mut Vec<u8>);
}

pub struct Bar;

impl Foo for Bar {
    fn foo(&self, v: &mut Vec<u8>) {
        // But this can't be implemented with a &mut[u8], so I need a &mut Vec<u8>
        v.pop();
    }
}

```

I saw this happen:

```
warning: writing `&mut Vec` instead of `&mut [_]` involves a new object where a slice will do
 --> src/main.rs:2:22
  |
2 |     fn foo(&self, v: &mut Vec<u8>);
  |                      ^^^^^^^^^^^^ help: change this to: `&mut [u8]`
  |
  = note: `#[warn(clippy::ptr_arg)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
```

I expected the code to be accepted

### Version

```text
rustc 1.60.0-nightly (e646f3d2a 2022-02-10)
binary: rustc
commit-hash: e646f3d2a9541952310778288854943678738ea9
commit-date: 2022-02-10
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 13.0.0
```

**Comment by Jarcho** [comment]
The fix was just merged up. It's fixed in 2022-02-11.

**Comment by little-dude** [comment]
I confirm that this is now fixed. Thank you!
